taxi/views.py

```diff
@@ -57,11 +76,24 @@ class ManufacturerDeleteView(LoginRequiredMixin, generic.DeleteView):
 class CarListView(LoginRequiredMixin, generic.ListView):
     model = Car
     paginate_by = 5
-    queryset = Car.objects.select_related("manufacturer")
+
+    def get_context_data(self, *, object_list=None, **kwargs) -> dict:
+        context = super(CarListView, self).get_context_data(**kwargs)
+        search = self.request.GET.get("search", "")
+        context["search_form"] = SearchForm(initial={"search": search})
+        return context
+
+    def get_queryset(self) -> QuerySet:
+        queryset = Car.objects.select_related("manufacturer")
+        filter_ = self.request.GET.get("search")
+        if filter_:
+            return queryset.filter(model__icontains=filter_)
+        return queryset
```

Comment:
1. Do you need to define class in super when only one parent class has this method?
2. Call super method to get an initial queryset
fix everywhere